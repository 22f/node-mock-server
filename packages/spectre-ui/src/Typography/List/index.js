// @flow

import "spectre.css";
import React, { Component } from "react";
import type { ListPropsType, ListEntryType } from "../../../spectre-ui.js.flow";

class List extends Component<ListPropsType> {
	// eslint-disable-next-line
	static displayName = "Typography/List";

	getItemTag(index: number): string {
		const { type = "ul" } = this.props;
		if (type === "dl") {
			return index % 2 === 0 ? "dt" : "dd";
		}
		return "li";
	}

	getKeyFromValue(value: string, index: number): string {
		return `${value.replace(/ /g, "").trim()}-${index}`;
	}

	get rootTag(): string {
		const { type = "ul" } = this.props;
		return `${type}`;
	}

	get className(): string {
		return this.props.className || "";
	}

	get classNameEntry(): string {
		return this.props.classNameEntry || "";
	}

	// eslint-disable-next-line
	renderEntry = (entry: ListEntryType, index: number): React$Element<*> => {
		const Tag = `${this.getItemTag(index)}`;
		const RootTag = this.rootTag;
		if (typeof entry === "object") {
			if (entry.entries.length >= 1) {
				return (
					<Tag key={this.getKeyFromValue(entry.name, index)} className={this.classNameEntry}>
						{entry.name}
						<RootTag className={this.className}>{entry.entries.map(this.renderEntry)}</RootTag>
					</Tag>
				);
			}
			return (
				<Tag key={this.getKeyFromValue(entry.name, index)} className={this.classNameEntry}>
					{entry.name}
				</Tag>
			);
		}
		return (
			<Tag key={this.getKeyFromValue(entry, index)} className={this.classNameEntry}>
				{entry}
			</Tag>
		);
	};

	render(): React$Element<*> {
		const { entries } = this.props;
		if (!(entries instanceof Array) || entries.length < 1) {
			return <div />;
		}
		const RootTag = this.rootTag;
		return <RootTag className={this.className}>{entries.map(this.renderEntry)}</RootTag>;
	}
}

export default List;
