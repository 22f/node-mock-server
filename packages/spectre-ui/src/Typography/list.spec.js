/* global describe it expect */

import React from "react";
import { shallow } from "enzyme";
import List from "./list";

const propsEntries = [{ name: "list item 2", entries: ["list item 2.1", "list item 2.2"] }, "list item 3"];

const props: Object = {
	entries: propsEntries,
	type: "ol"
};

describe("List", () => {
	describe("render", () => {
		it("snapshot: default", () => {
			const _props = { ...props };
			const tree = shallow(<List {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		it("snapshot: no type", () => {
			const _props = { ...props, type: undefined };
			const tree = shallow(<List {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		it("snapshot: type dl", () => {
			const _props = { ...props, type: "dl" };
			const tree = shallow(<List {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		it("snapshot: empty entries", () => {
			const _props = { ...props, entries: [] };
			const tree = shallow(<List {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		it("snapshot: empty entries in second level", () => {
			const entries = [].concat(propsEntries);
			entries[0].entries = [];
			const _props = { ...props, entries };
			const tree = shallow(<List {..._props} />);
			expect(tree).toMatchSnapshot();
		});
	});
});
