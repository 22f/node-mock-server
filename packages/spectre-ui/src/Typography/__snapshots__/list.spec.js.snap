// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List render snapshot: default 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ol
    className="class-name"
>
    <li
        className="class-name-entry"
    >
        list item 2
        <ol
            className="class-name"
        >
            <li
                className="class-name-entry"
            >
                list item 2.1
            </li>
            <li
                className="class-name-entry"
            >
                list item 2.2
            </li>
        </ol>
    </li>
    <li
        className="class-name-entry"
    >
        list item 3
    </li>
</ol>,
  "nodes": Array [
    <ol
      className="class-name"
>
      <li
            className="class-name-entry"
      >
            list item 2
            <ol
                  className="class-name"
            >
                  <li
                        className="class-name-entry"
                  >
                        list item 2.1
                  </li>
                  <li
                        className="class-name-entry"
                  >
                        list item 2.2
                  </li>
            </ol>
      </li>
      <li
            className="class-name-entry"
      >
            list item 3
      </li>
</ol>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <List
        className="class-name"
        classNameEntry="class-name-entry"
        entries={
                Array [
                        Object {
                          "entries": Array [
                            "list item 2.1",
                            "list item 2.2",
                          ],
                          "name": "list item 2",
                        },
                        "list item 3",
                      ]
        }
        type="ol"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": List {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "class-name",
          "classNameEntry": "class-name-entry",
          "entries": Array [
            Object {
              "entries": Array [
                "list item 2.1",
                "list item 2.2",
              ],
              "name": "list item 2",
            },
            "list item 3",
          ],
          "type": "ol",
        },
        "refs": Object {},
        "renderEntry": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ol
          className="class-name"
>
          <li
                    className="class-name-entry"
          >
                    list item 2
                    <ol
                              className="class-name"
                    >
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.1
                              </li>
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.2
                              </li>
                    </ol>
          </li>
          <li
                    className="class-name-entry"
          >
                    list item 3
          </li>
</ol>,
        "_debugID": 2,
        "_renderedOutput": <ol
          className="class-name"
>
          <li
                    className="class-name-entry"
          >
                    list item 2
                    <ol
                              className="class-name"
                    >
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.1
                              </li>
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.2
                              </li>
                    </ol>
          </li>
          <li
                    className="class-name-entry"
          >
                    list item 3
          </li>
</ol>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <List
    className="class-name"
    classNameEntry="class-name-entry"
    entries={
        Array [
            Object {
              "entries": Array [
                "list item 2.1",
                "list item 2.2",
              ],
              "name": "list item 2",
            },
            "list item 3",
          ]
    }
    type="ol"
/>,
}
`;

exports[`List render snapshot: empty entries 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div />,
  "nodes": Array [
    <div />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <List
        className="class-name"
        classNameEntry="class-name-entry"
        entries={Array []}
        type="ol"
/>,
      "_debugID": 11,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": List {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "class-name",
          "classNameEntry": "class-name-entry",
          "entries": Array [],
          "type": "ol",
        },
        "refs": Object {},
        "renderEntry": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 6,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div />,
        "_debugID": 12,
        "_renderedOutput": <div />,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <List
    className="class-name"
    classNameEntry="class-name-entry"
    entries={Array []}
    type="ol"
/>,
}
`;

exports[`List render snapshot: empty entries in second level 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ol
    className="class-name"
>
    <li
        className="class-name-entry"
    >
        list item 2
    </li>
    <li
        className="class-name-entry"
    >
        list item 3
    </li>
</ol>,
  "nodes": Array [
    <ol
      className="class-name"
>
      <li
            className="class-name-entry"
      >
            list item 2
      </li>
      <li
            className="class-name-entry"
      >
            list item 3
      </li>
</ol>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <List
        className="class-name"
        classNameEntry="class-name-entry"
        entries={
                Array [
                        Object {
                          "entries": Array [],
                          "name": "list item 2",
                        },
                        "list item 3",
                      ]
        }
        type="ol"
/>,
      "_debugID": 13,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": List {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "class-name",
          "classNameEntry": "class-name-entry",
          "entries": Array [
            Object {
              "entries": Array [],
              "name": "list item 2",
            },
            "list item 3",
          ],
          "type": "ol",
        },
        "refs": Object {},
        "renderEntry": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 7,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ol
          className="class-name"
>
          <li
                    className="class-name-entry"
          >
                    list item 2
          </li>
          <li
                    className="class-name-entry"
          >
                    list item 3
          </li>
</ol>,
        "_debugID": 14,
        "_renderedOutput": <ol
          className="class-name"
>
          <li
                    className="class-name-entry"
          >
                    list item 2
          </li>
          <li
                    className="class-name-entry"
          >
                    list item 3
          </li>
</ol>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <List
    className="class-name"
    classNameEntry="class-name-entry"
    entries={
        Array [
            Object {
              "entries": Array [],
              "name": "list item 2",
            },
            "list item 3",
          ]
    }
    type="ol"
/>,
}
`;

exports[`List render snapshot: no className 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ol
    className=""
>
    <li
        className="class-name-entry"
    >
        list item 2
        <ol
            className=""
        >
            <li
                className="class-name-entry"
            >
                list item 2.1
            </li>
            <li
                className="class-name-entry"
            >
                list item 2.2
            </li>
        </ol>
    </li>
    <li
        className="class-name-entry"
    >
        list item 3
    </li>
</ol>,
  "nodes": Array [
    <ol
      className=""
>
      <li
            className="class-name-entry"
      >
            list item 2
            <ol
                  className=""
            >
                  <li
                        className="class-name-entry"
                  >
                        list item 2.1
                  </li>
                  <li
                        className="class-name-entry"
                  >
                        list item 2.2
                  </li>
            </ol>
      </li>
      <li
            className="class-name-entry"
      >
            list item 3
      </li>
</ol>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <List
        className={undefined}
        classNameEntry="class-name-entry"
        entries={
                Array [
                        Object {
                          "entries": Array [
                            "list item 2.1",
                            "list item 2.2",
                          ],
                          "name": "list item 2",
                        },
                        "list item 3",
                      ]
        }
        type="ol"
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": List {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": undefined,
          "classNameEntry": "class-name-entry",
          "entries": Array [
            Object {
              "entries": Array [
                "list item 2.1",
                "list item 2.2",
              ],
              "name": "list item 2",
            },
            "list item 3",
          ],
          "type": "ol",
        },
        "refs": Object {},
        "renderEntry": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ol
          className=""
>
          <li
                    className="class-name-entry"
          >
                    list item 2
                    <ol
                              className=""
                    >
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.1
                              </li>
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.2
                              </li>
                    </ol>
          </li>
          <li
                    className="class-name-entry"
          >
                    list item 3
          </li>
</ol>,
        "_debugID": 4,
        "_renderedOutput": <ol
          className=""
>
          <li
                    className="class-name-entry"
          >
                    list item 2
                    <ol
                              className=""
                    >
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.1
                              </li>
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.2
                              </li>
                    </ol>
          </li>
          <li
                    className="class-name-entry"
          >
                    list item 3
          </li>
</ol>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <List
    className={undefined}
    classNameEntry="class-name-entry"
    entries={
        Array [
            Object {
              "entries": Array [
                "list item 2.1",
                "list item 2.2",
              ],
              "name": "list item 2",
            },
            "list item 3",
          ]
    }
    type="ol"
/>,
}
`;

exports[`List render snapshot: no classNameEntry 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ol
    className="class-name"
>
    <li
        className=""
    >
        list item 2
        <ol
            className="class-name"
        >
            <li
                className=""
            >
                list item 2.1
            </li>
            <li
                className=""
            >
                list item 2.2
            </li>
        </ol>
    </li>
    <li
        className=""
    >
        list item 3
    </li>
</ol>,
  "nodes": Array [
    <ol
      className="class-name"
>
      <li
            className=""
      >
            list item 2
            <ol
                  className="class-name"
            >
                  <li
                        className=""
                  >
                        list item 2.1
                  </li>
                  <li
                        className=""
                  >
                        list item 2.2
                  </li>
            </ol>
      </li>
      <li
            className=""
      >
            list item 3
      </li>
</ol>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <List
        className="class-name"
        classNameEntry={undefined}
        entries={
                Array [
                        Object {
                          "entries": Array [
                            "list item 2.1",
                            "list item 2.2",
                          ],
                          "name": "list item 2",
                        },
                        "list item 3",
                      ]
        }
        type="ol"
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": List {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "class-name",
          "classNameEntry": undefined,
          "entries": Array [
            Object {
              "entries": Array [
                "list item 2.1",
                "list item 2.2",
              ],
              "name": "list item 2",
            },
            "list item 3",
          ],
          "type": "ol",
        },
        "refs": Object {},
        "renderEntry": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ol
          className="class-name"
>
          <li
                    className=""
          >
                    list item 2
                    <ol
                              className="class-name"
                    >
                              <li
                                        className=""
                              >
                                        list item 2.1
                              </li>
                              <li
                                        className=""
                              >
                                        list item 2.2
                              </li>
                    </ol>
          </li>
          <li
                    className=""
          >
                    list item 3
          </li>
</ol>,
        "_debugID": 6,
        "_renderedOutput": <ol
          className="class-name"
>
          <li
                    className=""
          >
                    list item 2
                    <ol
                              className="class-name"
                    >
                              <li
                                        className=""
                              >
                                        list item 2.1
                              </li>
                              <li
                                        className=""
                              >
                                        list item 2.2
                              </li>
                    </ol>
          </li>
          <li
                    className=""
          >
                    list item 3
          </li>
</ol>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <List
    className="class-name"
    classNameEntry={undefined}
    entries={
        Array [
            Object {
              "entries": Array [
                "list item 2.1",
                "list item 2.2",
              ],
              "name": "list item 2",
            },
            "list item 3",
          ]
    }
    type="ol"
/>,
}
`;

exports[`List render snapshot: no type 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="class-name"
>
    <li
        className="class-name-entry"
    >
        list item 2
        <ul
            className="class-name"
        >
            <li
                className="class-name-entry"
            >
                list item 2.1
            </li>
            <li
                className="class-name-entry"
            >
                list item 2.2
            </li>
        </ul>
    </li>
    <li
        className="class-name-entry"
    >
        list item 3
    </li>
</ul>,
  "nodes": Array [
    <ul
      className="class-name"
>
      <li
            className="class-name-entry"
      >
            list item 2
            <ul
                  className="class-name"
            >
                  <li
                        className="class-name-entry"
                  >
                        list item 2.1
                  </li>
                  <li
                        className="class-name-entry"
                  >
                        list item 2.2
                  </li>
            </ul>
      </li>
      <li
            className="class-name-entry"
      >
            list item 3
      </li>
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <List
        className="class-name"
        classNameEntry="class-name-entry"
        entries={
                Array [
                        Object {
                          "entries": Array [
                            "list item 2.1",
                            "list item 2.2",
                          ],
                          "name": "list item 2",
                        },
                        "list item 3",
                      ]
        }
        type={undefined}
/>,
      "_debugID": 7,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": List {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "class-name",
          "classNameEntry": "class-name-entry",
          "entries": Array [
            Object {
              "entries": Array [
                "list item 2.1",
                "list item 2.2",
              ],
              "name": "list item 2",
            },
            "list item 3",
          ],
          "type": undefined,
        },
        "refs": Object {},
        "renderEntry": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="class-name"
>
          <li
                    className="class-name-entry"
          >
                    list item 2
                    <ul
                              className="class-name"
                    >
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.1
                              </li>
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.2
                              </li>
                    </ul>
          </li>
          <li
                    className="class-name-entry"
          >
                    list item 3
          </li>
</ul>,
        "_debugID": 8,
        "_renderedOutput": <ul
          className="class-name"
>
          <li
                    className="class-name-entry"
          >
                    list item 2
                    <ul
                              className="class-name"
                    >
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.1
                              </li>
                              <li
                                        className="class-name-entry"
                              >
                                        list item 2.2
                              </li>
                    </ul>
          </li>
          <li
                    className="class-name-entry"
          >
                    list item 3
          </li>
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <List
    className="class-name"
    classNameEntry="class-name-entry"
    entries={
        Array [
            Object {
              "entries": Array [
                "list item 2.1",
                "list item 2.2",
              ],
              "name": "list item 2",
            },
            "list item 3",
          ]
    }
    type={undefined}
/>,
}
`;

exports[`List render snapshot: type dl 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <dl
    className="class-name"
>
    <dt
        className="class-name-entry"
    >
        list item 2
        <dl
            className="class-name"
        >
            <dt
                className="class-name-entry"
            >
                list item 2.1
            </dt>
            <dd
                className="class-name-entry"
            >
                list item 2.2
            </dd>
        </dl>
    </dt>
    <dd
        className="class-name-entry"
    >
        list item 3
    </dd>
</dl>,
  "nodes": Array [
    <dl
      className="class-name"
>
      <dt
            className="class-name-entry"
      >
            list item 2
            <dl
                  className="class-name"
            >
                  <dt
                        className="class-name-entry"
                  >
                        list item 2.1
                  </dt>
                  <dd
                        className="class-name-entry"
                  >
                        list item 2.2
                  </dd>
            </dl>
      </dt>
      <dd
            className="class-name-entry"
      >
            list item 3
      </dd>
</dl>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <List
        className="class-name"
        classNameEntry="class-name-entry"
        entries={
                Array [
                        Object {
                          "entries": Array [
                            "list item 2.1",
                            "list item 2.2",
                          ],
                          "name": "list item 2",
                        },
                        "list item 3",
                      ]
        }
        type="dl"
/>,
      "_debugID": 9,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": List {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "className": "class-name",
          "classNameEntry": "class-name-entry",
          "entries": Array [
            Object {
              "entries": Array [
                "list item 2.1",
                "list item 2.2",
              ],
              "name": "list item 2",
            },
            "list item 3",
          ],
          "type": "dl",
        },
        "refs": Object {},
        "renderEntry": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 5,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <dl
          className="class-name"
>
          <dt
                    className="class-name-entry"
          >
                    list item 2
                    <dl
                              className="class-name"
                    >
                              <dt
                                        className="class-name-entry"
                              >
                                        list item 2.1
                              </dt>
                              <dd
                                        className="class-name-entry"
                              >
                                        list item 2.2
                              </dd>
                    </dl>
          </dt>
          <dd
                    className="class-name-entry"
          >
                    list item 3
          </dd>
</dl>,
        "_debugID": 10,
        "_renderedOutput": <dl
          className="class-name"
>
          <dt
                    className="class-name-entry"
          >
                    list item 2
                    <dl
                              className="class-name"
                    >
                              <dt
                                        className="class-name-entry"
                              >
                                        list item 2.1
                              </dt>
                              <dd
                                        className="class-name-entry"
                              >
                                        list item 2.2
                              </dd>
                    </dl>
          </dt>
          <dd
                    className="class-name-entry"
          >
                    list item 3
          </dd>
</dl>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <List
    className="class-name"
    classNameEntry="class-name-entry"
    entries={
        Array [
            Object {
              "entries": Array [
                "list item 2.1",
                "list item 2.2",
              ],
              "name": "list item 2",
            },
            "list item 3",
          ]
    }
    type="dl"
/>,
}
`;
